### PR TITLE
fix: index parent classes labels

### PR DESCRIPTION
**Pull request by gabrielfs7**
-- WIP --

Bring resources in the search if typing class label

https://oat-sa.atlassian.net/browse/ADF-15

### CODE REVIEW

File: models/classes/search/index/OntologyIndex.php [hunk 1]
```diff
@@ -15,26 +15,25 @@
...
 class OntologyIndex extends \core_kernel_classes_Resource
 {
-
-    const RDF_TYPE = "http://www.tao.lu/Ontologies/TAO.rdf#Index";
-    const PROPERTY_INDEX = 'http://www.tao.lu/Ontologies/TAO.rdf#PropertyIndex';
-    const PROPERTY_INDEX_FUZZY_MATCHING = 'http://www.tao.lu/Ontologies/TAO.rdf#IndexFuzzyMatching';
-    const PROPERTY_INDEX_IDENTIFIER = 'http://www.tao.lu/Ontologies/TAO.rdf#IndexIdentifier';
-    const PROPERTY_INDEX_TOKENIZER = 'http://www.tao.lu/Ontologies/TAO.rdf#IndexTokenizer';
-    const PROPERTY_DEFAULT_SEARCH = 'http://www.tao.lu/Ontologies/TAO.rdf#IndexDefaultSearch';
+    public const RDF_TYPE = "http://www.tao.lu/Ontologies/TAO.rdf#Index";
```
Comment (shpran) [inline models/classes/search/index/OntologyIndex.php:30]:
```suggestion
    public const RDF_TYPE = 'http://www.tao.lu/Ontologies/TAO.rdf#Index';
```

File: models/classes/search/tokenizer/ResourceClasses.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                $classes[] = $parentClass->getLabel();
+            }
+        }
+
+        return $classes;
+    }
+
+    private function hasReachedRootClass(core_kernel_classes_Class $class): bool
+    {
+        return in_array($class->getUri(), self::CLASSES_URIS_BLACK_LIST);
```
Comment (shpran) [inline models/classes/search/tokenizer/ResourceClasses.php:62]:
```suggestion
        return in_array($class->getUri(), self::CLASSES_URIS_BLACK_LIST, true);
```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/search/index/OntologyIndex.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/search/index/OntologyIndex.php:30]: Done

File: models/classes/search/tokenizer/ResourceClasses.php (hunk 2 above)
Comment (gabrielfs7) [reply models/classes/search/tokenizer/ResourceClasses.php:62]: Done

File: models/classes/search/tokenizer/ResourceClasses.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        'http://www.tao.lu/Ontologies/TAO.rdf#TAOObject',
+        'http://www.tao.lu/Ontologies/generis.rdf#generis_Ressource',
+        'http://www.w3.org/2000/01/rdf-schema#Resource'
+    ];
+
+    /**
+     * @param core_kernel_classes_Resource $resource
+     * @return array
+     */
+    public function getStrings($resource)
```
Comment (chinnu-focaloid) [inline models/classes/search/tokenizer/ResourceClasses.php:41]: `/**
* @param core_kernel_classes_Resource $resource
*/
public function getStrings($resource): array`

Comment (chinnu-focaloid) [inline models/classes/search/tokenizer/ResourceClasses.php:41]:
```suggestion
     */
    public function getStrings($resource): array
```

**Review by chinnu-focaloid — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/search/tokenizer/ResourceClasses.php (hunk 3 above)
Comment (gabrielfs7) [reply models/classes/search/tokenizer/ResourceClasses.php:41]: It is not respecting the interface declaration, so I kept like this :)

**Comment by AntonNeviarouski** [comment]
QA verified.

**Pull request merged**